plugins/DileptonCategories.cc

```diff
@@ -0,0 +1,126 @@
+#include <cp3_llbb/Framework/interface/MuonsProducer.h>
+#include <cp3_llbb/Framework/interface/ElectronsProducer.h>
+#include <cp3_llbb/Framework/interface/JetsProducer.h>
+
+#include <cp3_llbb/Framework/interface/DileptonCategories.h>
+#include <cp3_llbb/Framework/interface/DileptonAnalyzer.h>
+
+// ***** ***** *****
+// Dilepton Mu-Mu category
+// ***** ***** *****
+bool MuMuCategory::event_in_category_pre_analyzers(const ProducersManager& producers) const {
+    const MuonsProducer& muons = producers.get<MuonsProducer>("muons");
+    return muons.p4.size() >= 2 ;
+};
+
+bool MuMuCategory::event_in_category_post_analyzers(const ProducersManager& producers, const AnalyzersManager& analyzers) const {
+    const DileptonAnalyzer& dilepton_analyzer = analyzers.get<DileptonAnalyzer>("dilepton_analyzer");
+    return dilepton_analyzer.dimuons.size() > 0;
+};
+
+void MuMuCategory::register_cuts(CutManager& manager) {
+    manager.new_cut("ll_mass", "mll > 20");
+};
+
+void MuMuCategory::evaluate_cuts_pre_analyzers(CutManager& manager, const ProducersManager& producers) const {
+    const MuonsProducer& muons = producers.get<MuonsProducer>("muons");
+    if( (muons.p4[0] + muons.p4[1]).M() > 20.)
```

Comment:
Hi Olivier,
I don't understand this cut. One could imagine the cut being false, but while having the first(second) muon not passing muonID, and the second(first) and third muons passing muonID and having m_ll > 20. I this case we reject the event but we do have a good dimuon pair. 
Is there are reason for not keeping those events which I'm not seeing?